chain/src/txhashset.rs

```diff
@@ -1224,27 +1224,45 @@ fn input_pos_to_rewind(
 			head_header.height,
 			block_header.height
 		);
-		return Ok(bitmap);
+		return Ok(Bitmap::create());
 	}
 
-	//
-	// TODO - rework this loop to use Bitmap::fast_or() on a vec of bitmaps.
-	//
-	loop {
-		if current == block_header.hash() {
-			break;
+	let bitmap_fast_or = |b_res, block_input_bitmaps: &mut Vec<Bitmap>| -> Option<Bitmap> {
+		if let Some(b) = b_res {
+			block_input_bitmaps.push(b);
+			if block_input_bitmaps.len() < 256 {
+				return None;
+			}
 		}
+		let bitmap = Bitmap::fast_or(
+			&block_input_bitmaps
+				.iter()
+				.map(|x| x)
```

Comment:
I _think_ you can get rid of the `map(|x| x)` here and just call `.iter().collect()`.

Reply:
Ok, thanks, I learn a new trick, Yay!